### PR TITLE
Fix Nested group(chain(group)) fails

### LINKED ISSUE #5496: Nested group(chain(group)) fails
# Checklist

- [x] I have read the relevant section in the
  [contribution guide](http://docs.celeryproject.org/en/latest/contributing.html#other-bugs)
  on reporting bugs.
- [x] I have checked the [issues list](https://github.com/celery/celery/issues?q=is%3Aissue+label%3A%22Issue+Type%3A+Bug+Report%22+-label%3A%22Category%3A+Documentation%22)
  for similar or identical bug reports.
- [ ] I have checked the [pull requests list](https://github.com/celery/celery/pulls?q=is%3Apr+label%3A%22PR+Type%3A+Bugfix%22+-label%3A%22Category%3A+Documentation%22)
  for existing proposed fixes.
- [ ] I have checked the [commit log](https://github.com/celery/celery/commits/master)
  to find out if the bug was already fixed in the master branch.
- [x] I have included all related issues and possible duplicate issues
  in this issue (If there are none, check this box anyway).

## Mandatory Debugging Information

- [x] I have included the output of ``celery -A proj report`` in the issue.
    (if you are not able to do this, then at least specify the Celery
     version affected).
- [ ] I have verified that the issue exists against the `master` branch of Celery.
- [ ] I have included the contents of ``pip freeze`` in the issue.
- [ ] I have included all the versions of all the external dependencies required
  to reproduce this bug.

## Optional Debugging Information

- [x] I have tried reproducing the issue on more than one Python version
  and/or implementation.
- [ ] I have tried reproducing the issue on more than one message broker and/or
  result backend.
- [ ] I have tried reproducing the issue on more than one version of the message
  broker and/or result backend.
- [ ] I have tried reproducing the issue on more than one operating system.
- [ ] I have tried reproducing the issue on more than one workers pool.
- [ ] I have tried reproducing the issue with autoscaling, retries,
  ETA/Countdown & rate limits disabled.
- [ ] I have tried reproducing the issue after downgrading
  and/or upgrading Celery and its dependencies.

#### Related Issues

- None

#### Possible Duplicates

- Subsequent groups within a chain fail #3585
- Consecutive groups in chain fails #4848 
- task/group chains fails in some scenarios #5467

## Environment & Settings

**Celery version**: 4.3.0

<details>
<summary><b><code>celery report</code> Output:

software -> celery:4.3.0 (rhubarb) kombu:4.5.0 py:3.5.3
billiard:3.6.0.0 redis:3.2.1
platform -> system:Linux arch:64bit
kernel version:4.19.32-1-MANJARO imp:CPython
loader   -> celery.loaders.app.AppLoader
settings -> transport:redis results:redis://redis:6379/1

broker_url: redis://redis:6379/1
result_backend: redis://redis:6379/1

</b></summary>
<p>

```
```

</p>
</details>

# Steps to Reproduce
## Minimally Reproducible Test Case

<details>
<p>

```python
from celery import  group, chain
from tasks import task as t

# failing sequence
task = group([ t.si(),t.si(), chain( t.si(), group([ t.si(), t.si()]))])

# working sequence
task = group([ t.si(),t.si(), chain( t.si(), group([ t.si(), t.si()]), t.si())])

async_result = task.apply_async()
result = async_result.get()


```

</p>
</details>

# Expected Behavior

Calling `get` return group result. 

# Actual Behavior

All task finish success but calling `.get()` fail with traceback:

```
Traceback (most recent call last):
  File "/my_app.py", line 111, in add_to_queue
    result = async_result.get()
  File "/root/.cache/pypoetry/virtualenvs/my_app/lib/python3.7/site-packages/celery/result.py", line 697, in get
    on_interval=on_interval,
  File "/root/.cache/pypoetry/virtualenvs/my_app/lib/python3.7/site-packages/celery/result.py", line 815, in join_native
    on_message, on_interval):
  File "/root/.cache/pypoetry/virtualenvs/my_app/lib/python3.7/site-packages/celery/backends/asynchronous.py", line 137, in iter_native
    if node._cache:
AttributeError: 'GroupResult' object has no attribute '_cache'

```

### PR DESCRIPTION
**Pull request by tothegump**
- Fix #5496

### CODE REVIEW

**Comment by auvipy** [comment]
not sure why azure and dynamo db integration tests are failing

**Comment by thedrow** [comment]
I rebased and pushed it again.
Let's see if this was a temporary failure or not.

**Comment by thedrow** [comment]
I'm wondering if this always failed but we lacked the proper test case for it.
I'll check and report back.

**Comment by tothegump** [comment]
Oh, it's my fault, and the test case is right.
My test case is wrong. I think there is a bug of `Result` class in some condition.

**Comment by tothegump** [comment]
I will debug it, and fix it.

File: celery/concurrency/asynpool.py [hunk 1]
```diff
@@ -775,9 +775,6 @@ def _put_back(job, _time=time.time):
         def on_poll_start():
             # Determine which io descriptors are not busy
             inactive = diff(active_writes)
-            logger.debug(
```
Comment (auvipy) [inline celery/concurrency/asynpool.py]: should we must remove this?

Comment (tothegump) [reply celery/concurrency/asynpool.py]: Not must, but this log is extremely high frequency.
When I open debug level, I almost only see this log.
If this log is important, and should not delete, please let me know.

Comment (auvipy) [reply celery/concurrency/asynpool.py]: I think this could be reverted as not related. we can handle that in another issue.

Comment (tothegump) [reply celery/concurrency/asynpool.py]: Ok.

**Comment by auvipy** [comment]
Merging #5638 into master will decrease coverage by 0.04%.
The diff coverage is 23.07%.

**Comment by tothegump** [comment]
> Merging #5638 into master will decrease coverage by 0.04%.
> The diff coverage is 23.07%.

I thought `t/integration/test_canvas.py::test_chain::test_group_with_group_in_chain` covered the changes. Maybe I was wrong.
There is no unit test for `AsyncBackendMixin` and `ResultSet.join_native` before since `join_native` is mocked in unit test.
So, it may take some time to write correct unit test. Let me have a try.

**Comment by auvipy** [comment]
take some time. If you can't I will merge this as is?

**Comment by tothegump** [comment]
> 

> take some time. If you can't I will merge this as is?

OK, if you merge this, I will add unit test separately when I have enough time.

**Comment by thedrow** [comment]
@auvipy Please open an issue about this so we won't forget.

**Pull request merged**